makefile

```diff
@@ -1,5 +1,12 @@
-#CFLAGS=-Wextra -Wall -Wno-unused -pedantic -std=c99 -DNDEBUG -O3 
+UNAME := $(shell uname)
+
+ifeq ($(UNAME), Darwin)
+# Want to dynamically link on OS X
+CFLAGS=-Wextra -Wall -Wno-unused -pedantic -std=c99 -DNDEBUG -O3
```

Comment:
OS X supports statically linked libraries, but not executable binaries: https://developer.apple.com/library/mac/qa/qa1118/_index.html

It's been a while since I made this change, so I don't remember whether there was a specific snag I hit that prompted this